Makefile

```diff
@@ -379,6 +387,22 @@ unit-test: envtest
 	OPERATOR_NAMESPACE=$(OPERATOR_NAMESPACE) KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $(TEST_SRC) -v  -coverprofile cover.out
 CLEANFILES += cover.out
 
+.PHONY: extract-alert-rules
+extract-alert-rules: $(PROMETHEUS_CONFIG_YAML) 
+	./tests/scripts/extract_alerts.sh $(PROMETHEUS_CONFIG_YAML) $(GENERATED_ALERT_DIR)
+
+# Run prometheus-alert-unit-tests
+.PHONY: test-alerts
+test-alerts: extract-alert-rules
+	$(CONTAINER_RUNTIME) run --rm -t \
+	    -v "$(GENERATED_ALERT_DIR)":/prometheus/unit_tests:Z --entrypoint=/bin/sh \
+	    $(PROMETHEUS_IMAGE) -c 'cd unit_tests && promtool test rules *_unit_tests.yaml'
```

Comment:
any reason to run inside a container?
is it only because need promtool installed or if any other tools can be used from the image.
i would perfer to skip using a container for tests, locally should be fine, but then we invoke in GHA could be heavy

Reply:
I was running inside a container because, this test creates yaml files and a running it in a containers ensures it's not using the old files (by mistake). Should I still try skipping a container?

Reply:
i dont have strong opinion on it, even i will try to avoid start a container for this purpose, but i let @grdryn or Steven (where is Steven???) to weigh in

Reply:
The created files shouldn't be commited, and they are already ignored in the .gitignore, so I don't think using old files is a concern. We should add a cleanup step that automatically cleans any created files just in case if we are not running in a container but I don't think there's any reason other than requiring the install of promtool to necessitate running in a container, it's just a bit cleaner.